conf/messages.en

```diff
@@ -410,60 +410,116 @@ decoder.soe.ConsolidationOpen = the MUCR is still open, or contains no declarati
-error.ile.MucrNotShutDeparture = Cannot depart the MUCR from UK as MUCR is not Shut
-error.ile.FutureDateTimeOverExceeded = Message sent with a future date/time (greater than 25hours in the future)
-error.ile.UcrIsNotMucr = UCR is not for a MUCR
-error.ile.UcrNotExist = UCR to be disassociated does not exist
-error.ile.UcrAlreadyDisassociated = UCR to be disassociated is not already associated
-error.ile.UcrFieldCompletionNotMatchingEacAction = Master UCR/UCR field completion does not match the EAC action
-error.ile.EmptyMucr = MUCR is empty
-error.ile.InvalidExitMessage = Exit message failed validation/processing
-error.ile.LocationBasedPermissionFailed = Location-Based Permission failed
-error.ile.InvalidGoodsLocation = Goods Location is invalid
-error.ile.MucrAlreadyDeparted = MUCR has been departed
-error.ile.UcrRejectedUponArrival = UCR cannot be associated as it has been rejected upon arrival
-
-error.chief.MucrAlreadyShut = MUCR is already shut
-error.chief.UcrNotExist = Unique Consignment reference does not exist
-error.chief.Refused = Refused - User Role not matched on MUCR Arrivals
-error.chief.UcrAlreadyAssociated = UCR is already associated to Master
-error.chief.InvalidUcrFormat = Invalid characters in UCR or wrong length; please check Tariff
-error.chief.InvalidCharacterInUcrOrPart = Invalid character in UCR or Part
-error.chief.EmptyConsolidation = Shutting an empty consolidation is not permitted
-error.chief.CannotDepartMovements = No movements can be departed
-error.chief.DateTimeBeforeTimeout = Date / Time is before Timeout Period
-error.chief.DataValidationPrimaryError = Data validation primary error
-error.chief.InvalidCharacterInMessage = Message contains invalid characters
-error.chief.DeclarationFinalised = Action not permitted; Declaration is finalised
-error.chief.CancelledEntry = Entry has been cancelled
-error.chief.TooManyAnticipatedArrivals = Number of Anticipated Arrivals for UCR has been exceeded
-error.chief.AlreadyArrived = Already arrived by a Consolidation
-error.chief.MucrAlreadyInUseWithImports = Master UCR is already reserved for use with Imports consolidation
-error.chief.ArrivalAlreadyNotified = Anticipated arrival not allowed - Arrival already notified
-error.chief.InvalidMrnFormat = MRN: Format is invalid
-error.chief.MrnNotMatchEntry = MRN: Does not match the latest version on the Entry
-error.chief.CannotArriveContainedMucr = Unable to arrive MUCR as it is contained in another MUCR
-error.chief.CannotDepartContainedMucr = Unable to depart MUCR as it is contained in another MUCR
-error.chief.NotSuitableEntryReference = Entry Reference is not suitable for Export movement
-error.chief.MucrNotShutForAssociation = Only shut MUCRs can be associated into another MUCR
-error.chief.CannotShutMovingConsolidation = Cannot shut the consolidation since it is moving
+error.ile.InvalidUcrFormat.CDS = Invalid UCR Format, DUCR or MUCR
+error.ile.ClientIdValidationFailed.CDS = Client ID validation fails
```

Comment:
Still I'll think that in the file it will be easier. CSV with three columns `errorCode`, `exportsMessage`, `CDSMessage` message, load everything at the beginning of application and just based on the flag take CDS or Exports message, but it just different approach.

Approach with messages created manually can causes typos in message keys and it's more difficult to maintain.

Reply:
... but it's easier to translate (and we don't have to worry about it).   Do we have a test somewhere that every error code in the error code file has an entry in the messages file?

Reply:
yeah, that's true